File: backend/src/main/java/com/tododuk/domain/notification/dto/NotificationDto.java
package com.tododuk.domain.notification.dto;

import com.tododuk.domain.notification.entity.Notification;
import com.tododuk.domain.user.entity.User;
import lombok.NonNull;

public record NotificationDto(
        @NonNull int id,
        @NonNull User user,
        @NonNull String title,
        @NonNull String description,
        @NonNull String url,
        @NonNull boolean isRead
) {

    public NotificationDto(int id ,User user,String title, String description, String url, boolean isRead) {
        this.id = id;
        this.user = user;
        this.title = title;
        this.description = description;
        this.url = url;
        this.isRead = isRead;
    }

    public NotificationDto(Notification notification) {

        this(
                notification.getId(),
                notification.getUser(),
                notification.getTitle(),
                notification.getDescription(),
                notification.getUrl(),
                notification.isRead()
        );
    }

}

File: backend/src/main/java/com/tododuk/domain/notification/dto/NotificationDto.kt
package com.tododuk.domain.notification.dto


import com.tododuk.domain.notification.entity.Notification
import com.tododuk.domain.user.entity.User

data class NotificationDto(
    val id: Int,
    val user: User,  // Optional 대신 Kotlin nullable 타입 사용
    val title: String,
    val description: String,
    val url: String,
    val isRead: Boolean
) {
    companion object {
        fun from(notification: NotificationDto): NotificationDto {
            return NotificationDto(
                id = notification.id,
                user = notification.user,
                title = notification.title,
                description = notification.description,
                url = notification.url,
                isRead = notification.isRead
            )
        }
    }
    constructor(notification: Notification) : this(
        id = notification.id,
        user = notification.user,
        title = notification.title, // title이 null일 경우 기본값 설정
        description = notification.description,
        url = notification.url,
        isRead = notification.isRead
    )
}